### PR TITLE
Change Overrides to be just a function from Measure blk to Measure blk

### PR DESCRIPTION
**Pull request by EncodePanda**
No overrides is just simply an id function.
Also Overrides replaced MaxTxCapacityOverride.
Additionally it no longer comes as an argument to forgeBlock function
but is embedded in it.

### CODE REVIEW

File: ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Ledger.hs [hunk 1]
```diff
@@ -231,17 +228,14 @@ forgeDualByronBlock cfg curBlockNo curSlotNo tickedLedger maxTxCapacityOverride
       , dualBlockBridge = mconcat $ map vDualGenTxBridge vtxs
       }
   where
-    isNoMaxTxCapacityOverride = case maxTxCapacityOverride of
```
Comment (EncodePanda) [inline ouroboros-consensus-byron-test/src/Ouroboros/Consensus/ByronDual/Ledger.hs:234]: No `Eq` for functions 🤷

File: ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Node.hs [hunk 2]
```diff
@@ -76,8 +76,8 @@ data ByronLeaderCredentials = ByronLeaderCredentials {
       --
       -- Useful when the node is running with multiple sets of credentials.
     , blcLabel      :: Text
+    , blcOverrides  :: Overrides ByronBlock
     }
-  deriving (Show)
```
Comment (EncodePanda) [inline ouroboros-consensus-byron/src/Ouroboros/Consensus/Byron/Node.hs:80]: Hope they (Node) don't need it to have a `Show` instance

File: ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs [hunk 3]
```diff
@@ -257,6 +258,7 @@ protocolInfoShelleyBasedHardFork protocolParamsShelleyBased
           protocolParamsShelleyBased
           ()  -- trivial translation context
           protVer1
+          (Comp $ Overrides id)
```
Comment (EncodePanda) [inline ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs:261]: should be `Comp noOverride` instead

File: ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs [hunk 4]
```diff
@@ -289,6 +291,7 @@ protocolInfoShelleyBasedHardFork protocolParamsShelleyBased
             }
           transCtxt2
           protVer2
+          (Comp $ Overrides id)
```
Comment (EncodePanda) [inline ouroboros-consensus-cardano-test/src/Test/ThreadNet/Infra/ShelleyBasedHardFork.hs:294]: should be `Comp noOverride` instead

File: ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs [hunk 5]
```diff
@@ -234,58 +237,66 @@ data ProtocolParamsShelleyBased era = ProtocolParamsShelleyBased {
...
-                         protVer =
+                         protVer
+                         overrides =
     assertWithMsg (validateGenesis genesis) $
     ProtocolInfo {
         pInfoConfig       = topLevelConfig
       , pInfoInitLedger   = initExtLedgerState
       , pInfoBlockForging =
-          traverse (shelleyBlockForging tpraosParams) credentialss
+          traverse (flip (shelleyBlockForging tpraosParams) (singletonNP overrides)) credentialss
```
Comment (EncodePanda) [inline ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Node.hs:299]: second usage of `singletonNP` :)

File: ouroboros-consensus/src/Ouroboros/Consensus/Block/Forging.hs [hunk 6]
```diff
@@ -148,47 +149,12 @@ data BlockForging m blk = BlockForging {
...
-    -- of a block.
-    -- Compute maximum block transaction capacity
-    --
-    -- We allow the override to /reduce/ the maximum size, but not increase
-    -- it. This is important because any blocks exceeding the max block size
-    -- are invalid according to the ledger and we want certainly don't want to
-    -- forge invalid blocks.
-
-type family Overrides blk
+newtype Overrides blk = Overrides { unOverrides :: Measure blk -> Measure blk }
```
Comment (EncodePanda) [inline ouroboros-consensus/src/Ouroboros/Consensus/Block/Forging.hs:157]: We need those removed comments to be ported to `Overrides`. Also, maybe `Overrides` should be renamed back to `MaxTxCapacityOverride`?

File: ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Basics.hs [hunk 7]
```diff
@@ -136,20 +135,6 @@ newtype instance StorageConfig (HardForkBlock xs) = HardForkStorageConfig {
     }
   deriving newtype (NoThunks)
 
-{-------------------------------------------------------------------------------
-  Forging
--------------------------------------------------------------------------------}
-
-type instance Overrides (HardForkBlock xs) = NP MaxTxCapacityOverride xs
```
Comment (EncodePanda) [inline ouroboros-consensus/src/Ouroboros/Consensus/HardFork/Combinator/Basics.hs:143]: muhhaahahaha! :)

File: ouroboros-consensus-byron-test/test/Test/ThreadNet/Byron.hs [hunk 8]
```diff
@@ -1349,6 +1349,7 @@ updSignKey genSK extCfg cid eno newSK =
       , blcDlgCert    = newCert
       , blcCoreNodeId = cid
       , blcLabel      = "Updated Byron credentials"
+      , blcOverrides  = noOverride @ByronBlock
```
Comment (EncodePanda) [inline ouroboros-consensus-byron-test/test/Test/ThreadNet/Byron.hs:1352]: We should remove it from `Credentials`, that's not where `Overrides` belong

**Comment by nfrisby** [comment]
This was supplanted by PR #3261. Closing.

**Pull request closed without merging**